rust/kcl-lib/src/execution/kcl_value.rs

```diff
@@ -116,6 +116,10 @@ where
 }
 
 #[derive(Debug, Clone, PartialEq, Default)]
+// If you try to compare two `crate::std::StdFn` the results will be meaningless and arbitrary,
+// because they're just function pointers.
+// TODO: Add a newtype around crate::std::StdFn which manually impls PartialEq and sets it to false.
+#[allow(unpredictable_function_pointer_comparisons)]
```

Comment:
Calling this out for reviewer attention

Reply:
The purpose of this PR is to upgrade Rust, not to change behavior or fix possible issues.